tests/test_basic_sweeper.py

```diff
@@ -54,3 +58,42 @@ def test_split(
     )
     lret = [list(x) for x in ret]
     assert lret == expected
+
+
+def test_partial_failure(
+    monkeypatch: Any,
+    tmpdir: Any,
+) -> None:
+    monkeypatch.chdir("tests/test_apps/app_can_fail")
```

Comment:
since you are overriding the run_dir, I think you don't have to patch the chdir.
just run it from the Hydra root dir with full path.